### ISSUE TITLE
Economic Growth Analysis and Prediction

### ISSUE THREAD

**Issue by Vamshi-167** [Vamshi-167/Economic-Growth-Analysis-and-Prediction-#1]
# Economic Growth Analysis and Prediction

### 1.1 Project Overview

Economic growth is a fundamental indicator of a country’s economic health. This project aims to analyze historical economic growth data, identify key trends, and predict future economic conditions. By examining various economic indicators, we aim to provide actionable insights for policymakers and business leaders.

### 1.2 Objectives

- **Analyze** historical economic growth data to identify trends and patterns.
- **Understand** the relationship between economic growth and factors like GDP, inflation, and unemployment.
- **Build and Evaluate** predictive models to forecast future economic growth.

### 1.3 Scope and Limitations

- **Scope**: This project focuses on economic indicators from 2000 to 2020 for the US.
- **Limitations**: The accuracy of predictions is dependent on the quality of historical data and model assumptions.

---

### 2.1 Data Sources

- **World Bank Data**: [World Bank Economic Indicators](https://data.worldbank.org/)
- **IMF Data**: [IMF Economic Data](https://www.imf.org/en/Data)
- **OECD Data**: [OECD Economic Outlook](https://www.oecd.org/economic-outlook/)

### 2.2 Data Collection Methods

Data was collected from the above sources and saved in CSV format. Historical economic data, including GDP, inflation rates, and unemployment rates, was extracted and consolidated into a single dataset for analysis.

**Sample Data (`economic_data.csv`)**:
| Year | GDP     | Inflation | Unemployment |
|------|---------|-----------|--------------|
| 2000 | 10200   | 3.4       | 4.0          |
| 2001 | 10500   | 2.8       | 4.5          |
| ...  | ...     | ...       | ...          |
| 2020 | 21100   | 1.2       | 8.1          |

---

#### Summary Statistics

The dataset includes economic indicators from 2000 to 2020. Summary statistics show average GDP growth rates and inflation over this period.

#### Visualizations

- **GDP Over Time**: Shows how GDP has changed over the years.
- **Correlation Matrix**: Displays relationships between GDP, inflation, and unemployment.

**GDP Over Time**:

![Trend Plot](visualizations/trend_plot.png)

**Correlation Matrix**:

![Correlation Matrix](visualizations/correlation_matrix.png)

### 3.2 Data Cleaning

- **Handling Missing Values**: Missing values were imputed using forward fill.
- **Outlier Detection**: Outliers were identified using Z-score method and removed.
- **Data Transformation**: GDP growth rate was calculated as a percentage change.

### 3.3 Feature Engineering

- **New Features**: Added GDP growth rate.
- **Feature Selection**: Selected GDP growth rate as the primary feature for prediction.

---

### 4.1 Descriptive Analysis

Descriptive analysis revealed that GDP growth has been relatively stable, with periodic fluctuations correlating with economic cycles.

### 4.2 Correlation Analysis

Correlation analysis showed a significant relationship between GDP and unemployment, suggesting that higher GDP is associated with lower unemployment.

### 4.3 Predictive Modeling

- **Models Used**: Linear Regression, ARIMA for time-series forecasting.
- **Evaluation Metrics**: RMSE (Root Mean Square Error), MAE (Mean Absolute Error), R² score.

**Model Training and Evaluation**:
```python
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# Load data
data = pd.read_csv('data/cleaned_economic_data.csv')

# Feature selection
X = data[['GDP_growth_rate']]
y = data['GDP']

# Split data
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train model
model = LinearRegression()
model.fit(X_train, y_train)

# Predictions
y_pred = model.predict(X_test)

# Evaluation
rmse = mean_squared_error(y_test, y_pred, squared=False)
r2 = r2_score(y_test, y_pred)
print(f"RMSE: {rmse}, R²: {r2}")
```

### 4.4 Tools and Libraries

- **Python**: Programming language used.
- **Libraries**: Pandas, NumPy, Matplotlib, Seaborn, Scikit-learn, Statsmodels.

---

### 5.1 Key Findings

- **Trend**: GDP has generally increased over the years, with noticeable dips during economic recessions.
- **Correlation**: Strong correlation between GDP growth and unemployment rates.

### 5.2 Predictive Model Results

- **Model Performance**: The linear regression model achieved an R² score of 0.85 and an RMSE of 1200.
- **Forecasts**: Predictions suggest moderate GDP growth in the coming years.

**Forecast Plot**:

![Forecast Plot](visualizations/forecast_plot.png)

---

### 6.1 Summary

The analysis provided insights into historical economic growth patterns and identified key factors influencing growth. Predictive models indicate stable economic growth in the near future.

### 6.2 Recommendations

- **Policy Recommendations**: Consider policies to stabilize economic fluctuations and support sustained growth.
- **Business Strategy**: Use growth forecasts to plan for future market conditions.

### 6.3 Future Work

- Explore additional predictive models such as machine learning algorithms.
- Expand the analysis to include international comparisons.

---

### 7.1 Code Files

- **Data Preprocessing**: `scripts/data_preprocessing.py`
- **Exploratory Analysis**: `notebooks/exploratory_analysis.ipynb`
- **Predictive Modeling**: `scripts/modeling.py`

### 7.2 Instructions

1. **Setup**: Install required packages with `pip install -r requirements.txt`.
2. **Execution**: Run `data_preprocessing.py` to clean the data, followed by `exploratory_analysis.ipynb` for EDA, and `modeling.py` for predictive modeling.

---

## 8. References

- **Data Sources**:
  - World Bank. (n.d.). [World Bank Economic Indicators](https://data.worldbank.org/)
  - International Monetary Fund. (n.d.). [IMF Economic Data](https://www.imf.org/en/Data)
  - Organisation for Economic Co-operation and Development. (n.d.). [OECD Economic Outlook](https://www.oecd.org/economic-outlook/)
  
- **Tools and Libraries**:
  - Python Software Foundation. (n.d.). [Python Programming Language](https://www.python.org/)
  - Scikit-learn developers. (n.d.). [Scikit-learn Documentation](https://scikit-learn.org/stable/)

---

### 9.1 Supplementary Data

Additional tables and extended datasets are available upon request or included in the `data` folder.

### 9.2 Additional Visualizations

![Image](https://github.com/user-attachments/assets/fc1b55ba-d1a7-4cee-8e98-7307c5c13f2b)
![Image](https://github.com/user-attachments/assets/6f7a44ea-5654-40a1-a1e0-505c18a8a33b)
